docs/WIDGET_INTEGRATION_GUIDE.md

```diff
@@ -0,0 +1,929 @@
+    Note over AW: Widget displays<br/>selected asset name
+    Note over N: Node has new<br/>widget value
+    Note over C: Graph version<br/>incremented
+```
+
+**Key Communication Patterns:**
+
+1. **Props Flow Down**: ComponentWidgetImpl → Vue Component via `props`
+2. **Events Flow Up**: Vue Component → ComponentWidgetImpl via configured `setValue`
+3. **Canvas Context**: ComponentWidgetImpl manages LiteGraph integration
+4. **Value Updates**: BaseWidget handles the actual value change with proper context
+5. **UI Reactivity**: Vue components update automatically via reactive getters
+
+## Widget Replacement Pattern
+
+### 1. Create Widget Composable (`useAssetComboWidget.ts`)
+
+**✅ WORKING PATTERN:**
+```typescript
+export const useAssetComboWidget = (): ComfyWidgetConstructorV2 => {
+  const standardComboWidget = useComboWidget()
+  
+  return (node: LGraphNode, inputSpec: InputSpec) => {
+    // Eligibility check
+    const shouldUseAssetBrowser = checkAssetBrowserEligibility(node, inputSpec)
+    
+    if (shouldUseAssetBrowser) {
+      return createAssetPickerWidget(node, inputSpec)
+    }
+    
+    // Fallback to standard widget
+    return standardComboWidget(node, inputSpec)
+  }
+}
+
+// Widget creation using ComponentWidgetImpl
+function createAssetPickerWidget(node: LGraphNode, inputSpec: ComboInputSpec) {
+  const widgetValue = ref<string>(inputSpec.default || '')
+  const widget = new ComponentWidgetImpl({
+    node,
+    name: inputSpec.name,
+    component: AssetPickerWidget,
+    inputSpec,
+    options: {
+      getValue: () => widgetValue.value,
+      setValue: (value: string) => {
+        widgetValue.value = value
+      },
+      widget: {
+        value: widgetValue.value,
+        name: inputSpec.name,
+        setValue: (value: string) => {
+          widgetValue.value = value
+        }
+      }
+    }
+  })
+  
+  addWidget(node, widget as BaseDOMWidget<object | string>)
+  return widget
+}
+```
+
+**❌ BROKEN APPROACHES:**
+- Direct DOM manipulation instead of ComponentWidgetImpl
+- Modifying existing widgets in-place
+- Intercepting at the UI level instead of widget system level
+
+### 2. Register Widget in System (`widgets.ts`)
+
+**✅ WORKING PATTERN:**
+```typescript
+export const ComfyWidgets: Record<string, ComfyWidgetConstructor> = {
+  // Replace COMBO with enhanced version
+  COMBO: transformWidgetConstructorV2ToV1(useAssetComboWidget()),
+  // ... other widgets
+}
+```
+
+**❌ BROKEN APPROACHES:**
+- Adding new widget types instead of replacing existing ones
+- Conditional registration based on settings (breaks consistency)
+
+### 3. Vue Component Integration
+
+**✅ WORKING MODAL PATTERN:**
+```vue
+<template>
+  <div class="asset-picker-widget">
+    <!-- Widget display -->
+    <div class="selected-asset-display">
+      <span class="asset-name">{{ displayName }}</span>
+      <Button @click="openAssetBrowser" />
+    </div>
+    
+    <!-- Modal wrapper - CRITICAL: Use PrimeVue Dialog -->
```

Comment:
Probably can remove this... that's more useful for building a modal than getting a working vue component